libcaf_core/caf/async/file.hpp

```diff
@@ -13,9 +13,11 @@
 namespace caf::detail {
 
 /// A generator that emits characters from a file.
+template <typename T>
```

Comment:
Style: we always use `class` to refer to template parameters.

```suggestion
template <class T>
```